### PR TITLE
Add new operator configuration options to enable or disable the creation of ServiceMonitor and PrometheusRule for the operator itself

### PR DESCRIPTION
**Pull request by andreasgerstmayr**
This adds an operator reconciler which reconciles the operator's manifests based on the operator configuration.
Currently it is used to enable or disable the ServiceMonitor and PrometheusRule of the operator itself.

Resolves: #491

### CODE REVIEW

**Review by iblancasa — APPROVED** [review]
Great job!

**Review by rubenvp8510 — APPROVED** [review]

File: internal/operatormanifests/servicemonitor.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import (
+	"fmt"
+
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	configv1alpha1 "github.com/grafana/tempo-operator/apis/config/v1alpha1"
+)
+
+func serviceMonitor(featureGates configv1alpha1.FeatureGates, namespace string) *monitoringv1.ServiceMonitor {
```
Comment (pavolloffay) [inline internal/operatormanifests/servicemonitor.go:12]: Shall we move this to a separate prometheus package (the rules as well)?

**Review by pavolloffay — COMMENTED** [review]
LGTM.

I would consider adding a subpackage (e.g. `prometheus`) in the `operatormanifest`  package

File: internal/operatormanifests/servicemonitor.go (hunk 1 above)
Comment (pavolloffay) [reply internal/operatormanifests/servicemonitor.go:12]: Or I would move the `operatormanifests` under manifest package as `operator`. That makes more sense to me

Comment (andreasgerstmayr) [reply internal/operatormanifests/servicemonitor.go]: done

**Review by pavolloffay — APPROVED** [review]

**Pull request merged**